src/main/java/wolox/training/controllers/BookController.java

```diff
@@ -15,4 +34,66 @@ public String greeting(
         model.addAttribute("name", name);
         return "greeting";
     }
+
+    @GetMapping
+    public ResponseEntity<List<Book>> getAllBooks() {
+        List<Book> response = bookRepository.findAll();
+        if (response.isEmpty()) {
+            return ResponseEntity.noContent().build();
+        }
+        return ResponseEntity.ok(response);
+    }
+
+    @GetMapping("/{id}")
+    public ResponseEntity<Book> getBookById(@PathVariable("id") Long id) {
+        Optional<Book> response = bookRepository.findById(id);
+        if(response.isPresent()) {
+            return ResponseEntity.ok(response.get());
+        } else {
+            throw new BookNotFoundException(MSG);
+        }
```

Comment:
The benefit of using Optional is that you can avoid the if structure, like this: 

```
Book book = bookRepository.findById(id).orElseThrow(() -> new BookNotFoundException(MSG)); 
return ResponseEntity.ok(book); 

```

Let me know if you need help with this 😄  the same goes for the rest of the methods :)

Reply:
owww ok, thanks i will try by myself make the change if can't i will tell you